contracts/Tracer.sol

```diff
@@ -251,6 +251,75 @@ contract Tracer is ITracer, SimpleDex, Ownable {
         );
     }
 
+     /**
+    * @notice Match two orders that exist against each other
+    * @param order1 the first order that exists on chain
+    * @param order2 the second order that exists on chain
+    */
+    function matchOrders(uint order1, uint order2) public override {
+        uint256 fillAmount = _matchOrder(order1, order2);
+
+        int256 orderPrice = orders[order1].price;
+        address order1User = orders[order1].maker;
+        address order2User = orders[order2].maker;
+        bool order1Side = orders[order1].side;
+        int256 baseChange = (fillAmount.mul(uint256(orderPrice))).div(priceMultiplier).toInt256();
```

Comment:
I was going to write a comment about how there should be a check to make sure prices are the same, but then realised it is done in `SimpleDex._matchOrder`. I feel like there should be a comment above line 260 saying `// Perform compatibility checks and calculate fill amount` or something.

On the same now, should there be a check to make sure each order has different sides? At the moment it just checks one and assumes the other is the opposite.

Reply:
Agreed on the comment.  I want to do another critical sweep and really think where all the logic is sitting. What sits in Trader vs what sits in Tracer / SimpleDex.

Ah yes there should most definitely be a side. check.